src/ui/public/doc_table/components/table_row.js

```diff
@@ -158,6 +163,22 @@ module.directive('kbnTableRow', function ($compile) {
 
         return text;
       }
+
+      function truncateSourceSummary(sourceElement) {
```

Comment:
I thought about that, went back and forth a bit and I could still be convinced but here was my thinking:

The source formatter is a generic utility that could be used in lots of places, so while we want a truncated version in the doc table we might not want a truncated version everywhere the source formatter is used. Since it's the doc table that has specific requirements for the content, it's best suited to do the truncation. 

But I just realized the current implementation won't work. Someone could theoretically specify a different field formatter for the _source field, and this logic would break because it expects a particular html structure.

I thought about defining a new content type, so in addition to `text` and `html` maybe we'd have something like `html-summary` that the doc table could request but I dunno, it feels a little hacky, `html-summary` isn't really a content type. Do you have any ideas? I'm not super familiar with the field formatter code